src/Symfony/Component/Serializer/Encoder/CsvEncoder.php

```diff
@@ -41,6 +41,10 @@ class CsvEncoder implements EncoderInterface, DecoderInterface
      */
     public function __construct($delimiter = ',', $enclosure = '"', $escapeChar = '\\', $keySeparator = '.')
     {
+        if (3 > \func_num_args() && \PHP_VERSION_ID >= 70400) {
+            $escapeChar = '';
```

Comment:
would it make sense to turn the empty string to `'\\'` on < 7.4, so that ppl can start using it right now?

Reply:
the CSV output would still be broken on <7.4, as we implicitly change the escape-char... im not sure that's for the better :thinking:

Reply:
It would be consistent with what you're doing in https://github.com/symfony/symfony/pull/33744 and would save ppl from doing the same in their code. They could opt-on for the future (the empty string) and be ready. Alternatively, we could default to `null` and make null have this behavior instead of the empty string.

Reply:
yes, but in #33744 the user cannot pass the delimiter. So you get a fixed form of CSV parsing, based on your PHP version.

Here, the user might pass a delimiter. If so, we preserve it. If not, we calculate a default (so that's similar as the null approach, but using `null` is cleaner?).

If the user (explicitly) passes an empty string on <7.4 it's preserved and parsing crashes, as expected no?

Im really skeptical about changing the CSV output, without ppl updating/fixing the delimiter explicitly, as im unaware of the practical side effects. Letting uses change the delimiter makes them aware the CSV output may change also.

makes sense? So basically we might favor the `null` approach over `func_num_args()`, im fine either way :) but in 4.3 it's different anyway (not investigated yet).

Reply:
updated, i think you're right. We favor `''` as escape_char as of 3.4 for consistency, and use `'\\'` as a fallback on <7.4 internally.

this improves your CSV over time, as PHP gets updated.

I fail to see the complete behavior change, as i dont understand this escape_char thing really. All i know is this case `,"""","foo""","\""",\,foo\` :D which gets fixed (i assume php wont break previous valid CSV, which was my main concern)